clang/test/CodeGen/ubsan-aggregate-null-align.cpp

```diff
@@ -0,0 +1,134 @@
+// RUN: %clang_cc1 -triple x86_64-linux-gnu -fsanitize=alignment,null \
+// RUN:   -emit-llvm %s -o - | FileCheck %s --check-prefix=CHECK-UBSAN
+// RUN: %clang_cc1 -triple x86_64-linux-gnu -emit-llvm %s -o - \
+// RUN:   | FileCheck %s --check-prefix=CHECK-NO-UBSAN
+
+// Test that EmitAggregateCopy emits null and alignment checks when sanitizers
+// are enabled for aggregate copy operations with pointers.
+
+struct alignas(16) AlignedStruct {
+  int a;
+  int b;
+  int c;
+  int d;
+};
+
+struct NormalStruct {
+  int x;
+  int y;
+  int z;
+};
+
+// Stack-to-stack copies are optimized away (compiler knows they're valid)
```

Comment:
This needs `-NOT` check lines to actually test that the cases are optimized away; however, I don't think this is something we need to test in this PR because there should already be tests covering the underlying internal Clang APIs that we are calling. On top of the functional verification (that various specific checks are implemented), we only need enough to demonstrate that we are probably passing the right arguments to the APIs.